### PR TITLE
General: Add global ID to MVS::Image

### PR DESCRIPTION
**Pull request by rjanvier**
See #438 (and https://github.com/cdcseacave/openMVS/issues/438#issuecomment-504864399)

- Added the global `ID` to `MVS::Image`
- Added Global ID management in `Scene::LoadInterface(`) and `Scene::SaveInterface() `
- Tried to use the most plausible global `ID` in each interface.

### CODE REVIEW

File: libs/MVS/Image.h [hunk 1]
```diff
@@ -78,6 +78,7 @@ class MVS_API Image
 	uint32_t platformID; // ID of the associated platform
 	uint32_t cameraID; // ID of the associated camera on the associated platform
 	uint32_t poseID; // ID of the pose of the associated platform
+	uint32_t ID; // Global ID of the image
```
Comment (cdcseacave) [inline libs/MVS/Image.h]: pls initialize it to NO_ID

File: libs/MVS/Image.h [hunk 2]
```diff
@@ -119,21 +120,28 @@ class MVS_API Image
 	#ifdef _USE_BOOST
 	// implement BOOST serialization
 	template<class Archive>
-	void save(Archive& ar, const unsigned int /*version*/) const {
+	void save(Archive& ar, const unsigned int version) const {
 		ar & platformID;
 		ar & cameraID;
 		ar & poseID;
+		if (version > 2) {
```
Comment (cdcseacave) [inline libs/MVS/Image.h]: this is not enough, you need to inform boost of the current version, ex:
```
BOOST_CLASS_VERSION(MVS::Image, 3)
```
Also, current version is 0, so no need to set it to 3, as 1 is enough.

Pls note that `Interface.h` does not use boost, but a custom serialization in order to be portable. So do not confuse the 2, no connection between them.

Comment (rjanvier) [reply libs/MVS/Image.h]: thanks a lot! it should be resolved in the new commit.

**Comment by rjanvier** [comment]
Thanks for the review. This issue should be addressed in my last commit.

**Comment by cdcseacave** [comment]
LGTM, feel free to merge

**Comment by rjanvier** [comment]
I think I do not have the right to merge into your branch.

**Comment by cdcseacave** [comment]
hmm, ok, not sure how that works, so I'll just merge it

**Pull request merged**